### PR TITLE
PR2 | Generated System folder files (device.py, physical_channel.py, system.py)

### PR DESCRIPTION
**Pull request by MounikaBattu17**
**What does this Pull Request accomplish?**
- This PR generates System folder files (`device.py`, `physical_channel.py`, `system.py`)

**Why should this Pull Request be merged?**
This implements the 

- [Task 2316163](https://dev.azure.com/ni/DevCentral/_workitems/edit/2316163): Generate the device.py file
- [Task 2316168](https://dev.azure.com/ni/DevCentral/_workitems/edit/2316168): Generate the physical channel.py file
- [Task 2316155](https://dev.azure.com/ni/DevCentral/_workitems/edit/2316155): Generate the system.py

### CODE REVIEW

**Comment by MounikaBattu17** [comment]
The differences in the files are given below:

- `system.py `- has no changes
- `device.py` - The following properties were generated newly,
```
AI_PHYSICAL_CHANS
AO_PHYSICAL_CHANS
DI_LINES
DI_PORTS
DO_PORTS
DO_LINES
EXT_CAL_LAST_TEMP
SELF_CAL_LAST_TEMP
CAL_ACC_CONNECTION_COUNT
CAL_RECOMMENDED_ACC_CONNECTION_COUNT_LIMIT
CAL_USER_DEFINED_INFO_MAX_SIZE
SELF_CAL_SUPPORTED
CAL_DEV_TEMP
CAL_USER_DEFINED_INFO
CI_PHYSICAL_CHANS
CO_PHYSICAL_CHANS
EXT_CAL_RECOMMENDED_INTERVAL
```
But currently I have blacklisted these new properties as they were not present before.

- These are updates made in property name:
image

- physical_channel.py - these are the updates made in property name
image

File: generated/nidaqmx/system/physical_channel.py [hunk 1]
```diff
@@ -1180,33 +1180,62 @@ def teds_version_num(self):
...
         check_for_error(error_code)
 
-    def configure_teds(self, file_path=""):
+    def configure_teds(self, physical_channel="", file_path=""):
         """
         Associates TEDS information with the physical channel you
         specify. If you do not specify the filename of a data sheet in
-        the **file_path** input, this function attempts to find a TEDS
-        sensor connected to the physical channel. This function
+        the **physical_channel** input, this function attempts to find a
```
Comment (MounikaBattu17) [inline generated/nidaqmx/system/physical_channel.py:1228]: This change fix is already handled in Scrapigen metadata here https://github.com/ni/grpc-device-scrapigen/pull/150. Will have to update the` functions.py` here once the above PR finalize.

File: generated/nidaqmx/system/physical_channel.py [hunk 2]
```diff
@@ -1218,20 +1247,22 @@ def configure_teds(self, file_path=""):
...
+                        ctypes_byte_str, ctypes_byte_str, ctypes_byte_str]
 
         error_code = cfunc(
-            self._name, file_path)
+            self._name, physical_channel, file_path)
         check_for_error(error_code)
 
     def write_to_teds_from_array(
-            self, bit_stream=None,
+            self, physical_channel="", bit_stream=None,
```
Comment (MounikaBattu17) [inline generated/nidaqmx/system/physical_channel.py:1257]: Removing `device_name`fix is dependent on metadata functions.py from this https://github.com/ni/grpc-device-scrapigen/pull/150 and fix logic will be added once when that metadata finalize.

File: generated/nidaqmx/system/physical_channel.py [hunk 3]
```diff
@@ -1180,33 +1180,62 @@ def teds_version_num(self):
...
+        Args:
+            physical_channel (str): 
         """
         cfunc = lib_importer.windll.DAQmxClearTEDS
         if cfunc.argtypes is None:
             with cfunc.arglock:
                 if cfunc.argtypes is None:
                     cfunc.argtypes = [
-                        ctypes_byte_str]
+                        ctypes_byte_str, ctypes_byte_str]
```
Comment (zhindes) [inline generated/nidaqmx/system/physical_channel.py:1218]: this isn't right

```c
EXTERNC int32 __CFUNC DAQmxClearTEDS(const char physicalChannel[])
```

Comment (zhindes) [reply generated/nidaqmx/system/physical_channel.py:1218]: Looks like a few known issues like this you are resolving.

File: generated/nidaqmx/system/physical_channel.py [hunk 4]
```diff
@@ -1249,20 +1280,24 @@ def write_to_teds_from_array(
...
 
         error_code = cfunc(
-            self._name, bit_stream, len(bit_stream), basic_teds_options.value)
+            self._name, physical_channel, bit_stream, len(bit_stream),
+            basic_teds_options.value)
         check_for_error(error_code)
 
     def write_to_teds_from_file(
-            self, file_path="",
+            self, physical_channel="", file_path="",
```
Comment (zhindes) [inline generated/nidaqmx/system/physical_channel.py:1293]: fyi - this is wrong, too. I think you're fixing all of them, though.

File: generated/nidaqmx/system/physical_channel.py [hunk 5]
```diff
@@ -1180,33 +1180,62 @@ def teds_version_num(self):
...
+
+    @property
+    @deprecation.deprecated(deprecated_in="0.6.6", details="Use ci_supported_meas_types instead.")
+    def ci_meas_types(self):
+        return self.ci_supported_meas_types
+
+    @property
+    @deprecation.deprecated(deprecated_in="0.6.6", details="Use co_supported_output_types instead.")
+    def co_output_types(self):
+        return self.co_supported_output_types
```
Comment (zhindes) [inline generated/nidaqmx/system/physical_channel.py:1202]: I'm fine with all these

**Review by zhindes — CHANGES_REQUESTED** [review]

File: generated/nidaqmx/system/physical_channel.py (hunk 4 above)
Comment (MounikaBattu17) [reply generated/nidaqmx/system/physical_channel.py:1293]: Yes, it will fixed when metadata gets updated from Scrapigen

**Review by bkeryan — CHANGES_REQUESTED** [review]
Waiting on metadata update

File: generated/nidaqmx/system/device.py [hunk 6]
```diff
@@ -2513,32 +2574,38 @@ def reset_device(self):
...
         if cfunc.argtypes is None:
             with cfunc.arglock:
                 if cfunc.argtypes is None:
                     cfunc.argtypes = [
-                        ctypes_byte_str]
+                        ctypes_byte_str, ctypes_byte_str]
 
         error_code = cfunc(
-            self._name)
+            self._name, device_name)
```
[the remark this answers is not in the document]
Comment (MounikaBattu17) [reply generated/nidaqmx/system/device.py]: Resolved. updated `functions.py` from Scrapigen https://github.com/ni/grpc-device-scrapigen/pull/150.

File: generated/nidaqmx/system/physical_channel.py (hunk 1 above)
Comment (MounikaBattu17) [reply generated/nidaqmx/system/physical_channel.py]: Resolved.

File: generated/nidaqmx/system/physical_channel.py (hunk 2 above)
Comment (MounikaBattu17) [reply generated/nidaqmx/system/physical_channel.py]: Resolved.

File: generated/nidaqmx/system/physical_channel.py (hunk 3 above)
Comment (MounikaBattu17) [reply generated/nidaqmx/system/physical_channel.py]: Resolved.

File: generated/nidaqmx/system/physical_channel.py (hunk 4 above)
Comment (MounikaBattu17) [reply generated/nidaqmx/system/physical_channel.py]: Updated Scrapigen `functions.py`.

**Comment by MounikaBattu17** [comment]
> Waiting on metadata update

@zhindes @bkeryan  The `functions.py` metadata is updates from Scrapigen from this PR https://github.com/ni/grpc-device-scrapigen/pull/150 and resolve the open issues. Kindly re-review.

**Review by zhindes — APPROVED** [review]

File: src/codegen/utilities/attribute_helpers.py [hunk 7]
```diff
@@ -96,11 +117,92 @@
...
+        "deprecated_in": "0.6.6",
+    },
+    "field_daq_bank_devices": {
+        "new_name": "field_daq_bank_dev_names",
+        "deprecated_in": "0.6.6",
+    },
+    "field_daq_device": {
+        "new_name": "field_daq_dev_name",
+        "deprecated_in": "0.6.6",
+    },
```
Comment (bkeryan) [inline src/codegen/utilities/attribute_helpers.py:159]: These properties return `Device` objects, not names.

File: src/codegen/utilities/attribute_helpers.py [hunk 8]
```diff
@@ -96,11 +117,92 @@
...
+    "dev_serial_num": {
+        "new_name": "serial_num",
+        "deprecated_in": "0.6.6",
+    },
+    "tedshwteds_supported": {
+        "new_name": "teds_hwteds_supported",
+        "deprecated_in": "0.6.6",
+    },
```
Comment (bkeryan) [inline src/codegen/utilities/attribute_helpers.py:171]: @zhindes Should the new name be just `hwteds_supported`?

**Review by bkeryan — CHANGES_REQUESTED** [review]

File: src/codegen/utilities/attribute_helpers.py (hunk 8 above)
Comment (zhindes) [reply src/codegen/utilities/attribute_helpers.py:175]: Yep, let's do that.

File: src/codegen/utilities/attribute_helpers.py [hunk 9]
```diff
@@ -96,11 +117,92 @@
...
+    "co_output_types": {
+        "new_name": "co_supported_output_types",
+        "deprecated_in": "0.6.6",
+    },
+    "teds_hwteds_supported": {
+        "new_name": "tedshwteds_supported",
+        "deprecated_in": "0.6.6",
+    },
```
[the remark this answers is not in the document]
Comment (MounikaBattu17) [reply src/codegen/utilities/attribute_helpers.py]: Removed the incorrect deprecation key-value `teds_hwteds_supported`.

File: src/codegen/utilities/attribute_helpers.py (hunk 8 above)
Comment (MounikaBattu17) [reply src/codegen/utilities/attribute_helpers.py]: Changed `teds_hwteds_supported` to `hwteds_supported`.

**Comment by MounikaBattu17** [comment]
@zhindes - PR has got its required approvals. Kindly merge.

**Pull request merged**